engine/access/rpc/backend/connection_factory_test.go

```diff
@@ -33,6 +36,14 @@ func TestProxyAccessAPI(t *testing.T) {
 	connectionFactory := new(ConnectionFactoryImpl)
 	// set the collection grpc port
 	connectionFactory.CollectionGRPCPort = cn.port
+	// set the connection pool cache size
+	cache, _ := lru.New(5)
+	connectionFactory.ConnectionsCache = cache
+	// set metrics reporting
+	metrics := new(modulemock.TransactionMetrics)
+	metrics.On("ConnectionFromPoolRetrieved").Return(nil)
+	metrics.On("TotalConnectionsInPool", testifymock.Anything, testifymock.Anything).Return(nil)
```

Comment:
Rather than mocking the metrics, you can just use the `metrics.NewNoopCollector()`